ios/CodePush/CodePush.m

```diff
@@ -455,9 +455,12 @@ - (void)rollbackPackage
 {
     NSError *error;
     NSDictionary *failedPackage = [CodePushPackage getCurrentPackage:&error];
-
-    // Write the current package's metadata to the "failed list"
-    [self saveFailedUpdate:failedPackage];
+    if (error) {
```

Comment:
A non-nil error object doesn't always indicate that an error *actually* occurred.